testcases/kernel/syscalls/mount/mount06.c

```diff
@@ -38,126 +23,77 @@
-
-	for (lc = 0; TEST_LOOPING(lc); lc++) {
-
-		tst_count = 0;
-
-		SAFE_MOUNT(cleanup, device, mntpoint_src, fs_type, 0, NULL);
-
-		TEST(mount(mntpoint_src, mntpoint_des, fs_type, MS_MOVE, NULL));
-
-		if (TEST_RETURN != 0) {
-			tst_resm(TFAIL | TTERRNO, "mount(2) failed");
-		} else {
-
-			if (!ismount(mntpoint_src) && ismount(mntpoint_des))
-				tst_resm(TPASS, "move mount is ok");
-			else
-				tst_resm(TFAIL, "move mount does not work");
-
-			TEST(tst_umount(mntpoint_des));
-			if (TEST_RETURN != 0)
-				tst_brkm(TBROK | TTERRNO, cleanup,
-					 "umount(2) failed");
-		}
-	}
-
-	cleanup();
-	tst_exit();
-}
-
-int ismount(char *mntpoint)
-{
-	int ret = 0;
-	FILE *file;
-	char line[LINELENGTH];
-
-	file = fopen("/proc/mounts", "r");
-	if (file == NULL)
-		tst_brkm(TFAIL | TERRNO, NULL, "Open /proc/mounts failed");
-
-	while (fgets(line, LINELENGTH, file) != NULL) {
-		if (strstr(line, mntpoint) != NULL) {
-			ret = 1;
-			break;
-		}
-	}
-	fclose(file);
-	return ret;
-}
+static char mntpoint_dst[PATH_MAX];
 
 static void setup(void)
 {
-	tst_require_root();
-
-	tst_sig(NOFORK, DEF_HANDLER, cleanup);
-
-	tst_tmpdir();
-
-	fs_type = tst_dev_fs_type();
-	device = tst_acquire_device(cleanup);
-
-	if (!device)
-		tst_brkm(TCONF, cleanup, "Failed to obtain block device");
-
-	tst_mkfs(cleanup, device, fs_type, NULL, NULL);
-
-	if (getcwd(path_name, sizeof(path_name)) == NULL)
-		tst_brkm(TBROK, cleanup, "getcwd failed");
+	tmppath = tst_get_tmpdir();
 
 	/*
 	 * Turn current dir into a private mount point being a parent
 	 * mount which is required by move mount.
 	 */
-	SAFE_MOUNT(cleanup, path_name, path_name, "none", MS_BIND, NULL);
+	SAFE_MOUNT(tmppath, tmppath, "none", MS_BIND, NULL);
+	SAFE_MOUNT("none", tmppath, "none", MS_PRIVATE, NULL);
 
-	mount_flag = 1;
+	snprintf(mntpoint_src, PATH_MAX, "%s/%s", tmppath, MNTPOINT_SRC);
+	snprintf(mntpoint_dst, PATH_MAX, "%s/%s", tmppath, MNTPOINT_DST);
 
-	SAFE_MOUNT(cleanup, "none", path_name, "none", MS_PRIVATE, NULL);
+	SAFE_MKDIR(mntpoint_dst, 0750);
+}
 
-	snprintf(mntpoint_src, PATH_MAX, "%s/%s", path_name, MNTPOINT_SRC);
-	snprintf(mntpoint_des, PATH_MAX, "%s/%s", path_name, MNTPOINT_DES);
+static void cleanup(void)
+{
+	if (tst_is_mounted(mntpoint_src))
+		SAFE_UMOUNT(mntpoint_src);
 
-	SAFE_MKDIR(cleanup, mntpoint_src, DIR_MODE);
-	SAFE_MKDIR(cleanup, mntpoint_des, DIR_MODE);
+	if (tst_is_mounted(mntpoint_dst))
+		SAFE_UMOUNT(mntpoint_dst);
 
-	TEST_PAUSE;
+	if (tst_is_mounted(tmppath))
+		SAFE_UMOUNT(tmppath);
+
+	SAFE_RMDIR(mntpoint_dst);
 }
 
-static void cleanup(void)
+static void run(void)
 {
-	if (mount_flag && tst_umount(path_name) != 0)
-		tst_resm(TWARN | TERRNO, "umount(2) %s failed", path_name);
+	SAFE_MOUNT(tst_device->dev, mntpoint_src, tst_device->fs_type, 0, NULL);
+	SAFE_MOUNT(mntpoint_src, mntpoint_dst, tst_device->fs_type, MS_MOVE, NULL);
 
-	if (device)
-		tst_release_device(device);
+	TST_EXP_EXPR(
+		!tst_is_mounted(mntpoint_src) && tst_is_mounted(mntpoint_dst),
```

Comment:
Just a suggestion: This isn't a very strong check. It should probably try to access the files as in the previous test.